tests/regression/whole_axis_access.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <gridtools/common/integral_constant.hpp>
+#include <gridtools/sid/rename_dimension.hpp>
+#include <gridtools/stencil/cartesian.hpp>
+#include <gridtools/stencil/positional.hpp>
+
+#include <stencil_select.hpp>
+#include <test_environment.hpp>
+
+namespace {
+    using namespace gridtools;
+    using namespace stencil;
+    using namespace cartesian;
+
+    struct functor {
+        using out = inout_accessor<0>;
+        using in = in_accessor<1, extent<>, 4>;
+        using k_pos = in_accessor<2>;
+        using param_list = make_param_list<out, in, k_pos>;
+
+        template <class Eval>
+        GT_FUNCTION static void apply(Eval &&eval) {
+            auto &&res = eval(out());
+            auto k = eval(k_pos());
+            res = 0;
+            for (int kk = 0; kk < k; ++kk)
+                res += eval(in(0, 0, 0, kk));
```

Comment:
the only thing that bothers me a little is that the dimension to iterate over is the fourth and not the third. If the computation contained more than one stage that have to access the field `in` as regular field, I assume `in` should be passed twice to the computation: once as regular field and once as rename_dimension. Is that right?

Reply:
Answering the particular question: Yes. If the user needs two semantically different accesses to the data (absolute and relative indexing) he must pass this data twice. This is not that terrible performance wise -- only pointers are duplicated, the data is the same, the caching should work.

Answering generally: You should not expect the perfect solution for such a use case. Just because it is out of stencil computation bounds. However this test demonstrates the known feature of a good API: it is easy to code simple things; corner cases are still possible to implement reliably without backdoors and hacks. Another feature to demonstrate here is that our API is composable -- this solution uses the composition of several features: 1) accessors support offsets for the extra dimensions; 2) computation field interface is well defined by the SID concept; 3) universal (that works on any SID) SID transformations can be implemented with the zero performance cost.

Reply:
Thanks for the answer! I agree with you